examples/nlp/text_classification/conf/config.yaml

```diff
@@ -0,0 +1,81 @@
+# Sentence classification with pretrained BERT models
+
+io:
+  work_dir: ./outputs
+  checkpoint_dir: null
+  save_epoch_freq: 1
+  save_step_freq: -1
+
+pl:
+  trainer:
+    gpus: 1 # the number of gpus, 0 for CPU
+    num_nodes: 1
+    max_epochs: 10
+    max_steps: null # precedence over max_epochs
+    accumulate_grad_batches: 1 # accumulates grads every k batches
+    amp_level: O0 # O1/O2 for mixed precision
+    distributed_backend: ddp
+
+model:
+  data_dir: ??? # /path/to/data
+  class_balancing: null # or weighted_loss
+
+  train_ds:
+    prefix: train
+    batch_size: 32
+    shuffle: true
+    num_samples: -1
+    num_workers: 2
+    pin_memory: false
+    use_cache: true
+    drop_last: true
+
+  validation_ds:
+    prefix: dev
+    batch_size: 32
+    shuffle: false
+    num_samples: -1
+    num_workers: 2
+    pin_memory: false
+    use_cache: true
+    drop_last: true
+
+  language_model:
+    pretrained_model_name: roberta-base
+    max_seq_length: 36
+    bert_checkpoint: null
+    bert_config: null
+    tokenizer: null # only used if using custom checkpoint
+    tokenizer_model: null # only used if tokenizer is sentencepiece
+    do_lower_case: true # true for uncased models, false for cased models
+  
+  head:
+    num_output_layers: 2
+    fc_dropout: 0.1
+
+  optim:
+    name: adam
+    lr: 2e-5
+    args:
+      name: auto
```

Comment:
What is name:auto?

Reply:
Args is the static Config file that declares what configs are allowed inside the optimizers constructor (other than LR, which is mandatory). Since we would need to keep switching this config each time we switch the optimizer, I am using auto as a place holder and resolving the correct config file for the provided optimiser at runtime. 

Same with schedulers and their configs. For all intents and purposes, auto is good to have default. Only when final model is ready to release, should we think about making it static.

Reply:
https://github.com/NVIDIA/NeMo/pull/844#discussion_r454073078

Reply:
Such details are increasing learning curve for users. If there is not a way around avoiding them, it should be OK.